packages/tabs/src/LionTabs.js

```diff
@@ -0,0 +1,248 @@
+          visibility: visible;
+          display: block;
+        }
+
+        .tabs__panels {
+          display: block;
+        }
+      `,
+    ];
+  }
+
+  render() {
+    return html`
+      <div class="tabs__tab-group" role="tablist">
+        <slot name="tab"></slot>
+      </div>
+      <div class="tabs__panels">
+        <slot name="panel"></slot>
+      </div>
+    `;
+  }
+
+  constructor() {
+    super();
+    this.selectedIndex = 0;
+  }
+
+  firstUpdated() {
+    super.firstUpdated();
+    this.__setupSlots();
+  }
+
+  __setupSlots() {
+    const tabSlot = this.shadowRoot.querySelector('slot[name=tab]');
+    const handleSlotChange = () => {
+      this.__cleanStore();
+      this.__setupStore();
+      this.__updateSelected();
+    };
+    tabSlot.addEventListener('slotchange', handleSlotChange);
+  }
+
+  __setupStore() {
+    this.__store = [];
+    const buttons = this.querySelectorAll('[slot="tab"]');
+    const panels = this.querySelectorAll('[slot="panel"]');
+    if (buttons.length !== panels.length) {
+      // eslint-disable-next-line no-console
+      console.warn(
+        'The amount of tabs (',
+        buttons.length,
+        ") doesn't match the amount of panels (",
+        panels.length,
+        ').',
+      );
+    }
+
+    buttons.forEach((button, index) => {
+      const uid = uuid();
+      const panel = panels[index];
+      const entry = {
+        uid,
+        button,
+        panel,
+        clickHandler: this.__createButtonClickHandler(index),
+        keydownHandler: this.__handleButtonKeydown.bind(this),
+      };
+      this.__setupPanel({ element: entry.panel, ...entry });
+      this.__setupButton({ element: entry.button, ...entry });
+      this.__store.push(entry);
+    });
+  }
+
+  __cleanStore() {
+    if (!this.__store) {
+      return;
+    }
+    this.__store.forEach(entry => {
+      this.__cleanButton(entry.button, entry.clickHandler, entry.keydownHandler);
+    });
+  }
+
+  __setupPanel({ element, uid }) {
+    element.setAttribute('id', `panel-${uid}`);
+    element.setAttribute('role', 'tabpanel');
+    element.setAttribute('aria-labelledby', `button-${uid}`);
+    this.__deselectPanel(element);
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  __selectPanel(element) {
+    element.setAttribute('selected', true);
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  __deselectPanel(element) {
+    element.removeAttribute('selected');
+  }
+
+  __setupButton({ element, uid, clickHandler, keydownHandler }) {
+    element.setAttribute('id', `button-${uid}`);
+    element.setAttribute('role', 'tab');
+    element.setAttribute('aria-controls', `panel-${uid}`);
+    element.addEventListener('click', clickHandler);
+    element.addEventListener('keyup', keydownHandler);
+    this.__deselectButton(element);
+  }
+
+  // eslint-disable-next-line class-methods-use-this
+  __cleanButton(element, clickHandler, keydownHandler) {
```

Comment:
also not blocking - I'm just curious

I assume you put these pure function in the class so you can override them at some point?
however `__foo` sort of means never override me... 

so if they are pure and truly private (e.g. extending classes should not be able to override them) then you could put them as actual separate functions [which would enforce it]

or you could make them "protected" `_foo` so it's ok for extending classes to modify them